libs/partners/ai21/langchain_ai21/contextual_answers.py

```diff
@@ -0,0 +1,76 @@
+from typing import (
+    Any,
+    Optional,
+    Type,
+    TypedDict,
+)
+
+from langchain_core.documents import Document
+from langchain_core.runnables import RunnableConfig, RunnableSerializable
+from typing_extensions import TypeAlias
+
+from langchain_ai21.ai21_base import AI21Base
+
+ANSWER_NOT_IN_CONTEXT_RESPONSE = "Answer not in context"
+
+
+class ContextualAnswerInput(TypedDict):
+    context: str
+    question: str
+
+
+class AI21ContextualAnswers(RunnableSerializable[ContextualAnswerInput, str], AI21Base):
+    class Config:
+        """Configuration for this pydantic object."""
+
+        arbitrary_types_allowed = True
+
+    @property
+    def InputType(self) -> TypeAlias:
+        """Get the input type for this runnable."""
+        return ContextualAnswerInput
+
+    @property
+    def OutputType(self) -> Type[str]:
+        """Get the input type for this runnable."""
+        return str
+
+    def _convert_input(self, input: ContextualAnswerInput) -> ContextualAnswerInput:
+        if input.get("context") is None or input.get("question") is None:
```

Comment:
what happens if these are empty strings? I would check the truth value and not if these are None